internal/pkg/cmd/prerunner.go

```diff
@@ -118,6 +120,35 @@ func (c *AuthenticatedCLICommand) GetKafkaREST() (*KafkaREST, error) {
 	return (*c.KafkaRESTProvider)()
 }
 
+func (c *AuthenticatedCLICommand) GetMetricsClient() (*ccloudv2.MetricsClient, error) {
```

Comment:
The KafkaREST style is kind of tedious, I think.

The on-prem Kafka commands use the following line: `c.PersistentPreRunE = prerunner.InitializeOnPremKafkaRest(c.AuthenticatedCLICommand)` (although you currently still need to get the client before you use it...)

It seems cleaner to do something along these lines: in commands which need a particular client, use `c.PersistenPreRunE` to call a prerunner function to set that client's field in `AuthenticatedCLICommand`. Then access the client in the style that we currently use for all non-KafkaREST clients (`c.V2Client.Blah()`).

This way, we don't initialize clients that aren't use, and we also don't need to call `GetClient` functions potentially multiple times for the same command.

Reply:
Yeah, agreed. Let's save this for a separate PR though, since it's a substantial rewrite. Since we completely overwrite the `PersistentPreRunE` field, we should try to chain functions to the existing prerun function. I started writing support for that in the past, so there should be some helper functions already set up for that!